### PR TITLE
VH-35 i18n setup

### PR DESCRIPTION
**Pull request by rdgfuentes**
- Setup i18n
- Implement useTranslation in all components
- Apply date formatting using i18nDate
- Fix failing tests

### CODE REVIEW

File: applications/vessel-history/src/features/profile/Profile.tsx [hunk 1]
```diff
@@ -51,6 +51,13 @@ const Profile: React.FC = (props): React.ReactElement => {
 
   const [activeTab, setActiveTab] = useState<Tab | undefined>(tabs?.[0])
 
+  const defaultPreviousNames = useMemo(() => {
+    return `+${vessel?.history.shipname.byDate.length} previous ${t(
+      `vessel.names` as any,
+      'names'
```
Comment (satellitestudiodesign) [inline applications/vessel-history/src/features/profile/Profile.tsx:57]: Check https://www.i18next.com/translation-function/plurals if you need to show the translation of "Name" vs "Names"

**Review by satellitestudiodesign — APPROVED** [review]

File: applications/vessel-history/public/locales/en/translations.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  },
+  "selects": {
+    "allSelected": "All",
+    "selected": "selected"
+  },
+  "vessel": {
+    "authorizationPlural": "authorizations",
+    "historyLabelByField": "{{label}} $t(common.historyFor) {{vesselName}}",
+    "noAuthorizations": "No authorizations found",
+    "plusPreviousValuesByField": "+{{quantity}} $t(common.previous) {{fieldLabel}}",
```
Comment (fertandil87) [inline applications/vessel-history/public/locales/en/translations.json:29]: you need to be so sure about this, I mean use parts of another translations, when this become big and you don't remember all the relations (also if you remember) and you need to modify the "common.previous" text maybe in only one language and the new word doesn't match in the places where is used

**Review by fertandil87 — CHANGES_REQUESTED** [review]

File: applications/vessel-history/public/locales/fr/flags.json [hunk 3]
```diff
@@ -0,0 +1,252 @@
+{
```
Comment (fertandil87) [inline applications/vessel-history/public/locales/fr/flags.json:1]: do we need translations in all the languages for this file?

Comment (rdgfuentes) [reply applications/vessel-history/public/locales/fr/flags.json:1]: yes, we do

File: applications/vessel-history/src/features/profile/components/InfoFieldHistory.tsx [hunk 4]
```diff
@@ -24,13 +25,17 @@ const InfoFieldHistory: React.FC<ListItemProps> = ({
...
 
-  const formatedDate = useCallback((date: string | null = null) => {
-    return date
-      ? [DateTime.fromISO(date, { zone: 'UTC' }).toLocaleString(DateTime.DATETIME_MED), 'UTC'].join(
-          ' '
-        )
-      : ''
-  }, [])
+  const defaultTitle = useMemo(() => {
+    return `${label} History for ${vesselName}`
```
Comment (fertandil87) [inline applications/vessel-history/src/features/profile/components/InfoFieldHistory.tsx:29]: translate this?

**Review by fertandil87 — CHANGES_REQUESTED** [review]

File: applications/vessel-history/src/features/profile/components/InfoFieldHistory.tsx (hunk 4 above)
Comment (rdgfuentes) [reply applications/vessel-history/src/features/profile/components/InfoFieldHistory.tsx:29]: It's the default value fallback when no translation is found.

File: applications/vessel-history/public/locales/en/translations.json (hunk 2 above)
Comment (rdgfuentes) [reply applications/vessel-history/public/locales/en/translations.json]: Yeah, makes sense. I'll change it

File: applications/vessel-history/src/features/profile/Profile.tsx (hunk 1 above)
Comment (rdgfuentes) [reply applications/vessel-history/src/features/profile/Profile.tsx]: Fixed, thanks!

**Pull request merged**